### ISSUE TITLE
A task should be assignable on a single date, period

### ISSUE THREAD

**Issue by alexsegura** [coopcycle/coopcycle-web#874]
There is a general design problem with `TaskList`/`TaskCollection`, and the `Task::assignTo()` method. 

**TL;DR** when a task has a time range that spans over several days, the system assigns it to the last day of the range ☹️

The problem is in `TaskListProvider`
It takes the `doneBefore` property to resolve the list where the task should be assigned. 
This is due to the fact that the `assignTo()` method triggers some "magic" behind the scenes (with Doctrine subscribers), basically to assign linked tasks & manage notifications. 

https://github.com/coopcycle/coopcycle-web/blob/e3570d52866ce8e582133fc0156a79f8e82f0167/src/AppBundle/Doctrine/EventSubscriber/TaskSubscriber/TaskListProvider.php#L22-L47

As a result, the task is assigned **BOTH** to the task list of the current day (if dragged on the dashboard) and the task list of the last day of the range. 

---

Fortunately, there are only a few places where `assignTo()` is used:
- [AppBundle\Action\Task\Assign](https://github.com/coopcycle/coopcycle-web/blob/e3570d52866ce8e582133fc0156a79f8e82f0167/src/AppBundle/Action/Task/Assign.php#L64)
- [AppBundle\Serializer\TaskNormalizer](https://github.com/coopcycle/coopcycle-web/blob/e3570d52866ce8e582133fc0156a79f8e82f0167/src/AppBundle/Serializer/TaskNormalizer.php#L94)

Actions need to be taken to not break the app, while enforcing integrity:
- [ ] Introduce a new date property for the `PUT /api/tasks/{id}/assign` endpoint
- [ ] Do not accept the `assignedTo` property in the `PUT /api/tasks/{id}` endpoint
- [ ] Redesign database to enforce tasks to be assigned on a single day
